doc/syntax/comments.rdoc

```diff
@@ -168,22 +169,46 @@ The method Module#const_set is not affected.
 In this mode, all values assigned to constants are made shareable.
 
   # shareable_constant_value: experimental_everything
-  FOO = Set.new[1, 2, {foo: []}] # => ok, since this is
-      # same as `Set.new[1, 2, {foo: [].freeze}.freeze].freeze`
+  FOO = Set[1, 2, {foo: []}]
+  # same as FOO = Ractor.make_sharable(...)
+  # OR same as `FOO = Set[1, 2, {foo: [].freeze}.freeze].freeze`
 
   var = [{foo: []}]
   var.frozen? # => false (assignment was made to local variable)
   X = var # => calls `Ractor.make_shareable(var)`
   var.frozen? # => true
 
-This mode is "experimental", because it might be too error prone,
-for example by deep-freezing the constants of an external resource
-which could cause errors:
+This mode is "experimental", because it might be error prone, for 
+example by deep-freezing the constants of an external resource which 
+could cause errors:
 
   # shareable_constant_value: experimental_everything
   FOO = SomeGem::Something::FOO
   # => deep freezes the gem's constant!
 
+We will revisit to consider removing "experimental_" or removing this 
+mode by checking usages before Ruby 3.1.
```

Comment:
This will be revisited before Ruby 3.1 to either allow `everything`, or on the contrary remove this mode.